src/main/java/org/truffleruby/core/rope/RopeOperations.java

```diff
@@ -154,7 +154,11 @@ public static String decodeRope(Rope value) {
         value = flatten(value);
 
         Encoding encoding = value.getEncoding();
-        Charset charset = encodingToCharsetMap.computeIfAbsent(encoding, EncodingManager::charsetForEncoding);
+        Charset charset = encodingToCharsetMap.get(encoding);
+        if (charset == null) {
+            charset = EncodingManager.charsetForEncoding(encoding);
+            encodingToCharsetMap.putIfAbsent(encoding, charset);
```

Comment:
Could you comment here why it is not necessary to load the actually stored charset (It's a value).